Fix run order of `didInsertParent` for children of already-set-up parents

reidab opened the pull request:
Fixes #10

miguelcobain commented on tests/integration/components/did-insert-parent-hook-test.js:
```diff
@@ -68,18 +77,19 @@ test('top-level parent and two children-parents', function(assert) {
-  let parentSpy = this.parentSpy = this.spy();
-  let childSpy = this.childSpy = this.spy();
-  let childParentSpy = this.childParentSpy = this.spy();
+test('top-level parent with children-parents after if', function(assert) {
```
Why did you change existing tests rather than adding a new one?

reidab replied:
I actually started out with it in a separate test. I moved it here because it wasn't testing a new thing, just clarifying of how the run order should work for parents that were already set up.

I'm happy to split it back out again if you prefer.

reidab replied:
@miguelcobain Ping. Would you like me to split this out into a separate test?

miguelcobain replied:
Sorry for the delayed response, @reidab. Yes please, that would help a lot for me to understand what is going on here!

miguelcobain commented on addon/mixins/child.js:
```diff
@@ -55,11 +55,18 @@ export default Mixin.create({
     }
   },
 
+  notifyParentOfInsert: on('didInsertElement', function() {
```
We should use `didInsertParent` instead of `on('didInsertParent',`. It makes the calling order more predictable and provides a better opportunity for subclasses to override the method.

reidab replied:
I went with `on('didInsertElement'…` explicitly to avoid subclasses being able to override this and break the chain. I'll swap it.

reidab commented:
I'd completely forgotten about this PR until I went to update another library I had that depended on it. Apologies!

I've updated the PR to split out the new tests into their own test case and reverted back the changes to existing tests.

reidab commented:
Ping. I've been delaying a library depending on this fix for many months now. Is there anything else I can do to move it forward?

reidab commented:
Just rebased the branch on the current master resolved conflicts.

Duder-onomy commented:
@miguelcobain I just made PR #19 that will fix the failures seen in this PR.

reidab commented:
The description in @Duder-onomy's PR #19 said that it fixed this, so this got auto-closed, but it hasn't actually been merged. 😢

Duder-onomy commented:
@miguelcobain This is still an issue, My PR accidentally closed this.. All my PR did was fix TravisCi. 
Can you re-open this? Or even better, merge it?

Duder-onomy commented:
@miguelcobain 	Can this be merged?

Duder-onomy commented:
@miguelcobain This is still an issue, i'm going to try and explain the issue the best I can. 

When the change was made to register children on `init()` instead of `didInsertElement()`, children of an already-setup parent ( the children were hidden behind some conditional ) are receiving `didInsertParent` on init, which breaks and code in the `didInsertParent` hook that relies on the element being available.

Duder-onomy commented:
I believe this is the same issue seen in #16 and pull requested in #17.

knownasilya commented:
Ping, just for visibility.

miguelcobain commented:
The new architecture of ember-composability-tools is a lot different.
I added a modified version of this test to the current test suite and it passed. I can't keep the test though, because it required some hacking of the `Node` component to make the test possible (Good ol' times when we could just `willRender=this.spy` and it would end up replacing the component's willRender 😅 ).

In any case, I believe this is fixed, according to my tests. Feel free to reopen if this issue is found.
